lib/msf/ui/console/command_dispatcher/modules.rb

```diff
@@ -887,6 +888,42 @@ def cmd_reload_all(*args)
             self.driver.run_single("banner")
           end
 
+          def cmd_reload_search_help
+            print_line "Usage: reload_search"
+            print_line
+            print_line "Rebuild the module search cache"
+            print_line
+          end
+
+          def cmd_reload_search(*args)
+            if args.length > 0
+              cmd_reload_search_help
+              return
+            end
+
+            base_store = File.join(Msf::Config.install_root, "db",
+                                   Msf::Modules::Metadata::Store::BaseMetaDataFile)
+            user_store = File.join(Msf::Config.config_directory, "store",
+                                   Msf::Modules::Metadata::Store::UserMetaDataFile)
+
+            update_base = File.writable?(base_store)
+
+            if update_base
```

Comment:
I would suggest removing mods to `base_store`, a standard dev env should not replace base.  While it would be nice to remove the automation job, it seems like a better process to simply not merge changes to the base from PRs and always have the automation update it consistently.

This would not impact usability because the user's store is really always used and just recreated from the base if missing.

Reply:
Also if the updated base file does get checked in accidentally with local modules reported in cache that could expose pending research or work not expected to become public.

Reply:
I guess what I really want is for the module search to become accurate after running this command. If I leave the generated store in place under ~/.msf4/ it is unclear if this has the desired outcome.

Reply:
If you just update the user version then it search should become accurate.

Reply:
To be clarify the user store is `~/.msf4/store/modules_metadata.json` and what search loads/consumes. 

The base_store is in `<framwork path>/db/modules_metadata_base.json` which is only used to create first user version when no `~/.msf4/store/modules_metadata.json` is found.